src/pages/settings/Wallet/WalletPage/CardDetails.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import * as Expensicons from '../../../../components/Icon/Expensicons';
+import MenuItemWithTopDescription from '../../../../components/MenuItemWithTopDescription';
+import Clipboard from '../../../../libs/Clipboard';
+import useLocalize from '../../../../hooks/useLocalize';
+import usePrivatePersonalDetails from '../../../../hooks/usePrivatePersonalDetails';
+import ONYXKEYS from '../../../../ONYXKEYS';
+
+const propTypes = {
+    /** Card number */
+    pan: PropTypes.string,
+
+    /** Card expiration date */
+    expiration: PropTypes.string,
+
+    /** 3 digit code */
+    cvv: PropTypes.string,
+
+    /** User's private personal details */
+    privatePersonalDetails: PropTypes.shape({
+        /** User's home address */
+        address: PropTypes.shape({
+            street: PropTypes.string,
+            city: PropTypes.string,
+            state: PropTypes.string,
+            zip: PropTypes.string,
+            country: PropTypes.string,
+        }),
+    }),
+};
+
+const defaultProps = {
+    pan: '',
+    expiration: '',
+    cvv: '',
+    privatePersonalDetails: {
+        address: {
+            street: '',
+            street2: '',
+            city: '',
+            state: '',
+            zip: '',
+            country: '',
+        },
+    },
+};
+
+function CardDetails({pan, expiration, cvv, privatePersonalDetails}) {
+    usePrivatePersonalDetails();
+    const privateDetails = privatePersonalDetails || {};
+    const address = privateDetails.address || {};
```

Comment:
Should we apply default props somehow for address?
```suggestion
    const address = privateDetails.address || defaultProps.privatePersonalDetails.address;
```

Reply:
I don't think we have to do it that way, but I might implement something to handle empty address in the formatting function

Reply:
Firstly, I would move the variable into `getFormattedAddress` function. It is being used only there. 
Secondly, I would use destructuring assignment like: `const {street = '', city, state, ...} = privateDetails.address`.

Reply:
There was a discussion on how to handle these values, I added some new logic to handle empty addresses, let me know what you think!

Reply:
Looks good!